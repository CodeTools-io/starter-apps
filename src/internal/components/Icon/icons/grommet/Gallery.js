import React from 'react';
import { Gallery } from 'grommet-icons';

export default function GalleryIcon({ ...props }) {
  return <Gallery {...props} />;
}
