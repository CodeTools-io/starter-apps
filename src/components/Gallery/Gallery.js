import React from 'react';
import { Card } from 'grommet';
import GridLayout from 'components/GridLayout';

export default function Gallery({
  children,
  keyPrefix = 'galleryItem',
  ...props
}) {
  const galleryItems = React.Children.toArray(children);
  return (
    <GridLayout {...props}>
      {galleryItems.map((galleryItem, index) => (
        <Card key={`${keyPrefix}-${index}`} background="white" height="small">
          {galleryItem}
        </Card>
      ))}
    </GridLayout>
  );
}
