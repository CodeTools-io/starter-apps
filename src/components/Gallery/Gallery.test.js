import React from 'react'
import Gallery from './Gallery'

describe('Gallery', () => {
  test('can render', () => {
    const { container } = renderComponent(
      <Gallery>
        <p>Content for the Gallery</p>
      </Gallery>
    )

    expect(container).toMatchSnapshot()
  })
})

