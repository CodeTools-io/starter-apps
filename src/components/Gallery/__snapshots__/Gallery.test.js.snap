// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery can render 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 bFwQzJ"
  >
    <div
      class="StyledGrid-sc-1wofa1l-0 hxvnVW"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 chfukR"
      >
        <p>
          Content for the Gallery
        </p>
      </div>
    </div>
  </div>
</div>
`;
